monitor/services/src/main/java/org/datacleaner/monitor/configuration/ComponentsCache.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.monitor.configuration;
+
+import org.datacleaner.monitor.server.components.ComponentHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Class ComponentsCache
+ * For caching and storing Components and configuration
+ *
+ * @since 24.7.15
+ */
+public class ComponentsCache {
+    private static final Logger logger = LoggerFactory.getLogger(ComponentsCache.class);
+    private static final long CHECK_INTERVAL = 5 * 60 * 1000;
+    private static final long CLOSE_TIMEOUT = 60 * 1000;
+
+    private TenantContextFactory _tenantContextFactory;
+
+    private ConcurrentHashMap<String, ComponentsCacheConfigWrapper> data = new ConcurrentHashMap<>();
```

Comment:
A waaaay easier and more fail-safe suggestion for caching: Use a Google Guava cache :-)